cmd/rm-main.go

```diff
@@ -121,17 +133,31 @@ func checkRmSyntax(ctx *cli.Context) {
 	isForce := ctx.Bool("force")
 	isRecursive := ctx.Bool("recursive")
 	isStdin := ctx.Bool("stdin")
-
+	isDangerous := ctx.Bool("dangerous")
+	isNamespaceRemoval := false
+	for _, url := range ctx.Args() {
+		_, path := url2Alias(url)
```

Comment:
What are all expected behaviors for below command line arguments?
```
1. mc rm mylocaldir
2. mc rm mylocaldir/
3. mc rm mylocalfile
4. mc rm c:
5. mc rm c:\
```

Reply:
➜  mc git:(rr2) mkdir mylocaldir
➜  mc git:(rr2) mc rm mylocaldir 
Removing `mylocaldir`.
➜  mc git:(rr2) mkdir mylocaldir
➜  mc git:(rr2) touch mylocaldir/testfile
➜  mc git:(rr2) ✗ mc rm mylocaldir 
Removing `mylocaldir`.
mc: <ERROR> Failed to remove `mylocaldir`. remove mylocaldir: directory not empty
➜  mc git:(rr2) ✗ mc rm mylocaldir 
Removing `mylocaldir`.
mc: <ERROR> Failed to remove `mylocaldir`. remove mylocaldir: directory not empty

Reply:
however, removing the isAliasURLDir() check will interfere with mc rm messaging in recursive removal. Which is why the check was introduced in the first place - for filtering out fs.
```➜  mc git:(rr2) ✗ mc rm --r --force mylocaldir 
mc: <ERROR> Full site removal requires --dangerous option. This operation removes all the buckets on your namespace. Please review carefully before performing this *DESTRUCTIVE* operation. ```

Reply:
@harshavardhana and @balamurugana , added the isAliasURLDir() check back and a comment on why it is needed.